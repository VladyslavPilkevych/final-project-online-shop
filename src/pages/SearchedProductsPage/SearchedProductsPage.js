/* eslint-disable max-len */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CardItem from '../../components/CardItem/CardItem';
import styles from './SearchedProductsPage.module.scss';
import { searchProducts } from '../../store/actionCreators/searchAC';

function SearchedProductsPage() {
  const dispatch = useDispatch();
  const searchedProducts = useSelector((state) => state.search.searchedProducts);

  const useLocalStorageSearchedProducts = () => {
    dispatch(searchProducts(localStorage.getItem('phrase')));
  };

  window.addEventListener('load', useLocalStorageSearchedProducts);

  return (
    <div>
      <h2 className={styles.productName}>
        Products
        {' '}
        (
        {searchedProducts.length}
        )
      </h2>
      <div className={styles.searchedProductsContainer}>
        {searchedProducts.map(({
          _id, name, currentPrice, imageUrls, quantity, itemNo, model,
        }) => (
          <div key={_id} className={styles.cardWrapper}>
            <CardItem key={_id} id={_id} name={name} currentPrice={currentPrice} img={imageUrls[0]} quantity={quantity} itemNo={itemNo} model={model} />
          </div>
        ))}
      </div>
    </div>
  );
}

export default SearchedProductsPage;