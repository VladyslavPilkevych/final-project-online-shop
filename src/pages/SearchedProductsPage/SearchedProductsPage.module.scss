@import '../../assets/styles/global.module.scss';

.searchedProductsContainer {
    display: flex;
    flex-wrap: wrap;
    margin: 0 auto;
    max-width: 1398px;
    align-items: center;
    justify-content: space-between;
}

.cardWrapper {
    max-width: 232px;
}

.productName {
    margin: 0 auto;
    max-width: 1398px;
    margin-top: 30px;
    font-family: $fonts;
    font-style: normal;
    font-weight: 600;
    font-size: 32px;
    line-height: 48px;
}

@media screen and (min-device-width: 1px) and (max-device-width: 1392px) {
    .searchedProductsContainer {
        align-items: center;
        justify-content: space-between;
    }
}