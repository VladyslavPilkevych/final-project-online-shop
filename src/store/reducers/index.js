import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
