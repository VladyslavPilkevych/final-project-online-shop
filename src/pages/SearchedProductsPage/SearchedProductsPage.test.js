import { render } from '@testing-library/react';
import SearchedProductsPage from './SearchedProductsPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <SearchedProductsPage />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot SearchedProductsPage should render', () => {
  test('Snapshot SearchedProductsPage is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});