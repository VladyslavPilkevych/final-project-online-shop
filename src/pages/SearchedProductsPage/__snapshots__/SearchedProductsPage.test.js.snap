// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SearchedProductsPage should render Snapshot SearchedProductsPage is rendered 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="productName"
    >
      Products (0)
    </h2>
    <div
      class="searchedProductsContainer"
    />
    <div
      class="advantagesWrapper"
    >
      <div
        class="advantagesCard"
      >
        <img
          alt="advantages-icon"
          class="advantagesIcon"
          src="account-icon.png"
        />
        <h3
          class="advantagesTitle"
        >
          Product Support
        </h3>
        <p
          class="advantagesText"
        >
          Up to 3 years on-site warranty available for your peace of mind.
        </p>
      </div>
      <div
        class="advantagesCard"
      >
        <img
          alt="advantages-icon"
          class="advantagesIcon"
          src="support-icon.png"
        />
        <h3
          class="advantagesTitle"
        >
          Personal Account
        </h3>
        <p
          class="advantagesText"
        >
          With big discounts, free delivery and a dedicated support specialist.
        </p>
      </div>
      <div
        class="advantagesCard"
      >
        <img
          alt="advantages-icon"
          class="advantagesIcon"
          src="saving-icon.png"
        />
        <h3
          class="advantagesTitle"
        >
          Amazing Savings
        </h3>
        <p
          class="advantagesText"
        >
          Up to 70% off new Products, you can be sure of the best price.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
